src/components/FlavorStash/Note.js

```diff
@@ -0,0 +1,230 @@
+    loading: PropTypes.any.isRequired,
+    note: PropTypes.object.isRequired,
+    userId: PropTypes.string.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      editingNote: false,
+      updatedNote: null
+    };
+
+    this.handleNoteSubmit = this.handleNoteSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    const { actions, flavorId } = this.props;
+
+    actions.requestNote({ flavorId });
+  }
+
+  addNoteIcon(id) {
+    const { editingNote } = this.state;
+
+    return (
+      <Fragment>
+        {!editingNote && (
+          <Button
+            className="button-animation"
+            size="sm"
+            onClick={() => this.handleNoteEditor(id)}
+          >
+            <FontAwesomeIcon icon="plus" size="sm" title="Add Flavor Note" />
+          </Button>
+        )}
+      </Fragment>
+    );
+  }
+
+  editNoteIcon(id) {
+    const { editingNote } = this.state;
+
+    return (
+      <Fragment>
+        {!editingNote && (
+          <Button
+            className="button-animation"
+            size="sm"
+            onClick={() => this.handleNoteEditor(id)}
+          >
+            <FontAwesomeIcon icon="pen" size="sm" title="Edit Flavor Note" />
+          </Button>
+        )}
+      </Fragment>
+    );
+  }
+
+  handleNoteEditor() {
+    this.setState({ editingNote: !this.state.editingNote });
+  }
+
+  handleNoteSubmit(values) {
+    const { actions } = this.props;
+    const { note, update } = values;
+
+    if (!note) {
+      actions.deleteNote(values);
+    } else if (update === true) {
+      actions.updateNote(values);
+    } else {
+      actions.createNote(values);
+    }
+    this.setState({ updatedNote: note || '' });
+
+    this.handleNoteEditor();
+  }
+
+  noteEditor() {
+    const { flavorId, note, userId } = this.props;
+    const { updatedNote } = this.state;
+    const update = Boolean(note?.[flavorId]?.note);
+
+    return (
+      <FinalForm
+        onSubmit={this.handleNoteSubmit}
+        initialValues={{
+          flavorId,
+          note: updatedNote || note?.[flavorId]?.note,
+          update,
+          userId
+        }}
+        render={({ handleSubmit, submitting, values }) => (
+          <Form noValidate onSubmit={handleSubmit}>
+            <Form.Row>
+              <Field name="note">
+                {({ input }) => (
+                  <Form.Group as={Col}>
+                    <InputGroup>
+                      <Form.Control
+                        {...input}
+                        as="textarea"
+                        placeholder="Flavor Note"
+                      />
+                    </InputGroup>
+                  </Form.Group>
+                )}
+              </Field>
+            </Form.Row>
+            {values?.note ? (
+              <Form.Row>
+                <h4>Preview:</h4>
+                <Col md="12">
+                  {
+                    /* eslint-disable-next-line no-sync */
+                    unified()
```

Comment:
My two cents, this shouldn't be done synchronously or inline. Maybe use an async function that updates local component state with the parsed result?

Reply:
I haven't been able to get this to work with async. I also haven't been able to find an async example, they all show using this method